src/main/scala/beam/cosim/helics/BeamFederate.scala

```diff
@@ -23,11 +23,14 @@ object BeamFederate {
   var beamFed = Option.empty[BeamFederate]
 
   def getInstance(beamServices: BeamServices): BeamFederate = this.synchronized {
-    if (beamFed.isEmpty) {
-      loadHelics
-      beamFed = Some(BeamFederate(beamServices))
+    beamFed match {
+      case Some(beamFederate) => beamFederate
+      case None =>
+        loadHelics
```

Comment:
is `loadHelics` a side-effect method? is case of yes consider adding parenthesis to it. Looks like unused variable

Reply:
its not a method. Its a lazy val which is called only one time.